src/controllers/expenses.controller.js

```
@@ -0,0 +1,113 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+// Get all expenses
+const getAll = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  let expenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    expenses = expenses.filter((expense) => expense.userId === +userId);
```

getAll does not validate the `userId` query parameter. If `userId` is passed as a non-numeric value the current code will coerce it to `NaN` and the filter `expense.userId === +userId` will silently return no results instead of returning HTTP 400. Per the requirements, query IDs/formats must be validated and return 400 for malformed values. Consider doing `const numUserId = Number(userId); if (userId && Number.isNaN(numUserId)) return res.status(400).send({ message: 'Write correct data' });` and then filter using the numeric `numUserId`.